cpp/src/io/utilities/datasource.cpp

```diff
@@ -27,30 +27,66 @@ namespace cudf {
 namespace io {
 
 /**
- * @brief Implementation class for reading from a file or memory source using
- * memory mapped access.
- *
- * Unlike Arrow's memory mapped IO class, this implementation allows memory
- * mapping a subset of the file where the starting offset may not be zero.
+ * @brief Base class for file input. Only implements direct device reads.
  */
-class memory_mapped_source : public datasource {
-  class memory_mapped_buffer : public buffer {
-    size_t _size   = 0;
-    uint8_t *_data = nullptr;
+class file_source : public datasource {
```

Comment:
If this is not meant to be used directly, and only through `memory_mapped_source` or `direct_read_source`, you may want to hide this in a namespace.

Reply:
I don't think this would be useful since the declaration is in the .cpp file. It's not visible in other files anyway.

Reply:
Technically it is visible to other files during link time. Putting this into anonymous namespace ensures it is only visible in this file. It also communicates to reviewers and future contributors that it is intended only for this file.